### ISSUE TITLE
Add name and description to left menu

### ISSUE THREAD

**Issue by atsansone** [dart-lang/site-www#5789]
Hotlist: https://buganizer.corp.google.com/hotlists/5891206
Bug: https://buganizer.corp.google.com/issues/337340353

### Observed Results

While navigating using VoiceOver or ChromeVox, can be noticed that the menu in the left side of the screen does not have a proper name or description.

* [Screencast Voiceover](http://screencast/cast/NDg1MDgyMTAxMDI5MjczNnwzMzVhZWI0OS0yOA)

### Expected Results

The dropdown buttons names should be appropriate and read a meaningful name.

### User Impact

The users can get confused since the buttons have a long name, not designed to be easy for users that rely more on hearing that sight.

### Steps To Reproduce

1. Enable VoiceOver or ChromeVox
1. Navigate to https://dart.dev/guide
1. Navigate the page until the left menu where the dropdown menus are is reached.
1. Notice that the names that are read aloud are not simplified for users that rely mostly in hearing support.

### Relevant GAR Sheet

[A11Y Tracker](https://a11ytracker.corp.google.com/products/034ba691-ac78-4cc2-88c2-e06449cfd17e/edit/testresults?useCase=5uc12pk41e1hg&e=ProductLabels)

### GAR Details

[GAR 1.11](go/GAR-web#purpose-link-control-button)

### Environment Details

* Machine: Macbook/ Chromebook
* Browser: Version 124.0.6367.62 (Official Build) (arm64)
* Screen reader: VoiceOver/ChromeVox
* Bug filling time: 15 mins
